lib/valueflow.cpp

```diff
@@ -6646,11 +6646,11 @@ static void valueFlowForLoopSimplifyAfter(Token* fortok, nonneg int varid, const
         endToken = fortok->scope()->bodyEnd;
 
     Token* blockTok = fortok->linkAt(1)->linkAt(1);
-    std::list<ValueFlow::Value> values;
-    values.emplace_back(num);
-    values.back().errorPath.emplace_back(fortok,"After for loop, " + var->name() + " has value " + values.back().infoString());
-
     if (blockTok != endToken) {
+        std::list<ValueFlow::Value> values;
+        values.emplace_back(num);
+        values.back().errorPath.emplace_back(fortok,"After for loop, " + var->name() + " has value " + values.back().infoString());
```

Comment:
There is an overload of valueFlowForward that just takes a single value and you don't need to put it into a list.

Reply:
Will adjust and check the other calls as well. Just moved the existing code.

Reply:
Those cleanups got rid of the things GCC 4.8 cannot handle. Really nice. 😀